apps/desktop-wallet/src/components/Ellipsed.tsx

```diff
@@ -28,33 +28,30 @@ interface EllipsedProps extends HTMLAttributes<HTMLDivElement> {
 
 const Ellipsed = ({ text, className }: EllipsedProps) => {
   const el = useRef<HTMLDivElement | null>(null)
-  const charWidth = useRef<number>()
   const [_text, setText] = useState(text)
 
   const handleResize = useCallback(() => {
-    if (el?.current === null) return
-
-    if (charWidth.current === undefined) {
-      charWidth.current = el.current.scrollWidth / text.length
+    if (el?.current === null) {
+      setText(text.substring(0, 5) + '...')
+    } else {
+      const visibleChars = Math.floor(el.current.clientWidth / (el.current.scrollWidth / text.length))
+      const half = visibleChars / 2
+
+      setText(
+        visibleChars >= text.length
+          ? text
+          : text.slice(0, Math.floor(half) - 2) +
+              (visibleChars === text.length ? '' : '...') +
+              text.slice(-Math.ceil(half) + 2)
+      )
```

Comment:
It seems that this was a pain in the ass... I just did a quick googling and found this. WDYT? Worth the try?

https://www.npmjs.com/package/react-middle-ellipsis

Reply:
We used it in the past but the perfs were bad...

Reply:
ah ok!